ui/app/App.js

```diff
@@ -1298,6 +1334,29 @@ class App extends React.Component {
       NotebookSections.get(id).source = code;
     };
 
+    const runGcode = async ({ e, path, data, workspace }) => {
+      const { model } = this.state;
+      await this.Notebook.clickMake(e);
+      // Find the active Cnc (if any).
+      const makeNode = model.getNodeById('Make');
+      const subModel = makeNode.getExtraData().model;
+      const tabsetNode = subModel.getNodeById('Make/Tabset');
+      const selected = tabsetNode.getSelected();
+      if (selected === -1) {
+        return;
+      }
+      const tabNode = tabsetNode.getChildren()[selected];
+      if (tabNode.id === 'Make/Tabset/Tab') {
+        return;
+      }
+      const { ip, type } = tabNode.getConfig();
+      const cnc = getCnc({ ip, type });
+      if (path && !data) {
+        data = await readOrWatch(path, { workspace });
+      }
+      await cnc.uploadAndRun(data);
```

Comment:
🥇

Reply:
I don't think there are any security implications.

The CNC side hasn't changed, and the browser is sending blind, and listening for feedback from the websocket.

It just makes the conversations a bit more awkward.

Reply:
Couldn't some random website take control of your CNC and turn it on?